From b1f63be6d2fe8821e32b84e3c11c804f5b474a0f Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Sun, 16 Nov 2025 11:47:49 +0100
Subject: [PATCH] build: Use debug option to detect development versions

meson's buildtype option contains not only if the build is not using debug,
but also configures the optimization option.

Ref: https://mesonbuild.com/Builtin-options.html#details-for-buildtype
Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 src/meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index a3824a228363..41b3ec1b2f24 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -108,11 +108,11 @@ cargo_bin = find_program('cargo')
 cargo_opt = ['--manifest-path', meson.project_source_root() / 'Cargo.toml']
 cargo_opt += ['--target-dir', meson.project_build_root() / 'src']
 
-if get_option('buildtype') == 'release'
+if get_option('debug')
+  cargo_target = 'debug'
+else
   cargo_opt += ['--release']
   cargo_target = 'release'
-else
-  cargo_target = 'debug'
 endif
 
 cargo_env = environment()
-- 
2.51.2

